### PR TITLE
TSEN-2383: Expose disconnect/reconnect channel, return consumerTag, c…

### PR DESCRIPTION
**Pull request by vukicevic**
…leanup code

### CODE REVIEW

**Comment by joegoldbeck** [comment]
Doesn't refer the changes in this review, but I think the pattern of creating a connection to Rabbit immediately upon requiring the library is a fragile one.

**Comment by vukicevic** [comment]
@joegoldbeck I didn't want to modify existing behavior, I am not sure if we are using this module already, if we are that would be a breaking change (not connecting automatically).

File: RabbitMQ.js [hunk 1]
```diff
@@ -263,15 +231,14 @@ function createConnection() {
...
-        .catch( (err) => {
-            Promise
-                .delay(retryInterval)
-                .then(() => {
-                    createConnection(); // try to reconnect again if it fails
-                    retry++;
-                    debug(`Retry RabbitMQ connection: ${retry} attempts. Error: ${err.message}`);
-                })
-        });
+        .catch(err => Promise
```
Comment (joegoldbeck) [inline RabbitMQ.js]: This one seems odd. Is there a better way other than nesting the promise?

If it's left as is, I think there is a style issue to fix. I believe Airbnb suggest wrapping in parens.

https://github.com/airbnb/javascript#arrows--paren-wrap

**Comment by vukicevic** [comment]
Added .eslintrc and eslint-config-tetrascience, worked through files to fix errors.

File: RabbitMQ.js [hunk 2]
```diff
@@ -172,53 +147,42 @@ class RabbitMQ extends EventEmitter{
...
+            .then(channel => {
                 channel.prefetch(this.prefetch);
-                channel.consume(queueName, callback);
+                return channel.consume(this.queueName, callback);
             })
-            .catch((err) => {
-                debug(`Failed to set message callback on queueName ${this.queueName}. Error: ${err.message}`);
-            });
-
+            .then(response => this.callbacks[response.consumerTag] = callback)
```
Comment (spinwang) [inline RabbitMQ.js]: `this.callbacks` is an array I think. Based on http://www.squaremobius.net/amqp.node/channel_api.html#channel_consume , `cosumerTag` is not an integer.

**Comment by spinwang** [comment]
Nice change! left a comment regarding `.then(response => this.callbacks[response.consumerTag] = callback)`

File: RabbitMQ.js (hunk 2 above)
Comment (vukicevic) [inline RabbitMQ.js]: Good catch, this was a code change from an earlier commit I reverted, but didn't cleanup everywhere, fixed in new commit.

**Pull request merged**